prime-router/docs/standard-operating-procedures/Poison-queue/poison-queue-management.md

```diff
@@ -0,0 +1,29 @@
+## Context
+
+Each step in both the Universal Pipeline has built in logic that retries the function five times
+before moving the failing message to a poison queue. The exceptions to this rule are the "Receive" step which has no retry logic
+and the "Send" step which has custom retry logic. More information on retry logic for the send step can be found in the Universal Pipeline [send step documentation](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/docs/universal-pipeline/send.md)
+
+## Poison Queues
+![Azure-queues](Azure-queues.png)
+
+## Resolving Messages in the Poison Queue
+
+When a message ends up in a poison queue it will remain there for seven days before being deleted. Within this seven day period the underlying issue needs to be resolved
+and the message resent or else the items the report will have to be resent manually through the pipeline (If within the 60 day retention window). If the underlying issue has been
+fixed and merged, or the issue was transient and does not require a code change the messages in the poison queue can be resent through Azure Storage Explorer.
+
+## Resending from Azure Storage Explorer
+Login in to Azure Storage Explorer with an account with elevated privilages ("SU" account). Within Azure storage explorer, navigate to the relevant queue by finding the storage container for the appropriate environment (prod or staging) in the navigation window on the left hand side of Azure Storage explorer. Navigate to queues and then double click on the queue name you need to resolve messages from.
+This will bring up the queue and allow you to view all messages waiting in the queue for resolution.
+
+![translate-poison-queue](translate-poison-queue.png)
+
+Select one or more messages and click the "Move Messages" option at the top of the window. This brings up a list of all queue that the messages can be moved to.
+Be sure to select the queue that matches the pipeline step that the messages failed in (ex: messages in elr-fhir-translate-poison should be moved to elr-fhir-translate, messages from elr-fhir-convert-poison should be moved to elr-fhir-convert etc.)
+
```

Comment:
Is the only way to know that messages processed successfully is that they don't end up in the poison queue? or should we check the messages get routed somewhere if they need to?

Reply:
If they failed again they would throw errors again and then end up in the poison queue again. A lot of the messages (at least right now) won't actually go anywhere in the UP and I'm not sure how you would check where they were "supposed" to go without manually looking through potentially thousands of messages.

Reply:
Yeah, there's no way to know without look manually looking. It just seems weird to wait and see if the message doesn't end up in the poison queue again, but whoever fixes the issue should probably test these messages locally and make sure they can get processed.

Reply:
Agreed. Other than the timeout issue though it should fail immediately.